### PR TITLE
Pandas fix

### PR DESCRIPTION
**Pull request by Pizza2Pizza**
Updated versions of pandas, pint and pint-pandas:

pandas = "^2.0"       
pint = "^0.24"
pint-pandas = "0.6"

Updated functions due to internal changes of pint-pandas that distinguish between pandas Series with object data type and one with pint units

----
📚 Documentation preview 📚: https://portfolyo--99.org.readthedocs.build/en/99/

### CODE REVIEW

File: portfolyo/tools/freq.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 # Allowed frequencies.
-ALLOWED_FREQUENCIES_DOCS = "'15T' (=quarterhour), 'H', 'D', 'MS', 'QS' (or 'QS-FEB', 'QS-MAR', etc.), or 'AS' (or 'AS-FEB', 'AS-MAR', etc.)"
+ALLOWED_FREQUENCIES_DOCS = "'15T' (=quarterhour), 'H', 'D', 'MS', 'QS' (or 'QS-FEB', 'QS-MAR', etc.), or 'YS' (or 'AS-FEB', 'AS-MAR', etc.)"
```
Comment (rwijtvliet) [inline portfolyo/tools/freq.py:12]: YS-FEB, YS-MAR

File: portfolyo/tools/unit.py [hunk 2]
```diff
@@ -160,3 +160,48 @@ def split_magn_unit(
...
+        # We may have a series of pint quantities. Convert to pint-series, if possible.
+        dimensions = {v.dimensionality for v in fr.values}
+        if len(dimensions) != 1:
+            raise ValueError(
+                f"Expected a Series with quantities of the same dimension; got {dimensions}."
+            )
+        # Convert all values to same unit.
+        units = fr.values[0].units
+        magnitudes = [v.to(units).magnitude for v in fr.values]
+        return pd.Series(magnitudes, fr.index, dtype=f"pint[{units}]")
```
[the remark this answers is not in the document]
Comment (rwijtvliet) [reply portfolyo/tools/unit.py:204]: @Pizza2Pizza

File: docs/specialized_topics/dataprep.rst [hunk 3]
```diff
@@ -71,7 +71,7 @@ Frequency
 
 The index must have a frequency (``fr.index.freq``) and it must be a valid frequency. To check validity of the frequency, one may use ``portfolyo.assert_freq_valid()``. The following abbreviations are used by ``pandas`` and throughout this package:
 
-* ``15T``: quarterhourly; 
+* ``15min``: quarterhourly; 
 * ``H``: hourly;
```
Comment (rwijtvliet) [inline docs/specialized_topics/dataprep.rst:75]: should this be lowercase?

File: portfolyo/core/pfstate/pfstate.py [hunk 4]
```diff
@@ -209,8 +209,8 @@ def asfreq(self, freq: str = "MS") -> PfState:  # from ABC
         Parameters
         ----------
         freq : str, optional
-            The frequency at which to resample. 'AS' for year, 'QS' for quarter, 'MS'
-            (default) for month, 'D for day', 'H' for hour, '15T' for quarterhour.
+            The frequency at which to resample. 'YS' for year, 'QS' for quarter, 'MS'
+            (default) for month, 'D for day', 'H' for hour, '15min' for quarterhour.
```
Comment (rwijtvliet) [inline portfolyo/core/pfstate/pfstate.py:213]: @Pizza2Pizza: should 'H' be 'h'?

File: portfolyo/tools/freq.py [hunk 5]
```diff
@@ -251,8 +251,8 @@ def longest(*freqs: str) -> str:
 
     Examples
     --------
-    >>> freq.longest('MS', 'H', 'AS', 'D')
-    'AS'
+    >>> freq.longest('MS', 'H', 'YS', 'D')
```
Comment (rwijtvliet) [inline portfolyo/tools/freq.py:254]: @Pizza2Pizza 'h'?

File: portfolyo/tools/freq.py [hunk 6]
```diff
@@ -231,7 +231,7 @@ def shortest(*freqs: str) -> str:
 
     Examples
     --------
-    >>> freq.shortest('MS', 'H', 'AS', 'D')
+    >>> freq.shortest('MS', 'H', 'YS', 'D')
```
Comment (rwijtvliet) [inline portfolyo/tools/freq.py:234]: @Pizza2Pizza 'h'?

File: docs/specialized_topics/dataprep.rst (hunk 3 above)
Comment (Pizza2Pizza) [reply docs/specialized_topics/dataprep.rst:75]: yes

https://github.com/pydata/xarray/issues/8394

Comment (Pizza2Pizza) [reply docs/specialized_topics/dataprep.rst:75]: oi, you meant 'H'
also, yes

**Review by rwijtvliet — APPROVED** [review]

**Pull request merged**